web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -76,13 +76,14 @@ export class DataStoreService {
 
   // TODO: Define return types for methods in this class
   updateLayer(projectId: string, layer: Layer) {
-    const { id: layerId, forms, ...layerDoc } = layer;
+    const { id: layerId, name, forms, ...layerDoc } = layer;
 
     return this.db
       .collection('projects')
       .doc(projectId)
       .update({
         [`layers.${layerId}`]: {
+          name: name?.toJS() || {},
```

Comment:
Is this necessary? I believe it's already included in `layerDoc`?

Reply:
yes, name is a StringMap(Map<string, string>), it needs a `toJS()` for the same reason `forms` needs `toJS()`

Reply:
Oooic.. Thanks for reminding!